[WebRTC] Extract BaseTarget-libaom.xcconfig from libaom.xcconfig

ddkilzer opened the pull request:
#### f67a9fb1e5fff1909406a6c3e7f6d19261f10e41
<pre>
[WebRTC] Extract BaseTarget-libaom.xcconfig from libaom.xcconfig
<a href="https://bugs.webkit.org/show_bug.cgi?id=267022">https://bugs.webkit.org/show_bug.cgi?id=267022</a>
&lt;<a href="https://rdar.apple.com/120399649">rdar://120399649</a>&gt;

Reviewed by Elliott Williams.

* Source/ThirdParty/libwebrtc/Configurations/BaseTarget-libaom.xcconfig: Add.
- Extract settings from libaom.xcconfig.
(WARNING_CFLAGS):
- Add missing $(inherited) to define warning flags consistently.
- Add -Wno-unused-function to fix the build after adding $(inherited).
* Source/ThirdParty/libwebrtc/Configurations/libaom.xcconfig:
- Include BaseTarget-libaom.xcconfig after extracting settings.
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
- Add BaseTarget-libaom.xcconfig to the project.

Canonical link: <a href="https://commits.webkit.org/272641@main">https://commits.webkit.org/272641@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e9573ccd1d6c4b5491fb2ad0ec02d328ba9f1904

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32315 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11052 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34136 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/34869 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29258 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13403 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8258 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/28808 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/32733 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9316 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/28878 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/8110 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/8250 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/28809 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36210 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29378 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29239 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/34375 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8384 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6325 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32237 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10039 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9015 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4197 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/8944 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple reviewed (commented):
Why have a `BaseTarget-libaom` config that's only used once, by `libaom`?

emw-apple commented on Source/ThirdParty/libwebrtc/Configurations/BaseTarget-libaom.xcconfig:
```diff
@@ -0,0 +1,45 @@
+HEADER_SEARCH_PATHS[arch=x86_64] = $(inherited) Source/third_party/libaom/source/config/linux/generic Source/third_party/libaom/source/libaom Source/third_party/libaom/source/config;
+HEADER_SEARCH_PATHS[arch=arm64*] = $(inherited) Source/third_party/libaom/source/config/linux/arm64-cpu-detect Source/third_party/libaom/source/libaom Source/third_party/libaom/source/config;
+
+USE_HEADERMAP = NO;
```
FWIW, I know this setting is deprecated, and it might even be a no-op these days? No need to do anything about it now, but if you notice that it's safe to remove as you set up new targets, please feel free to.

ddkilzer replied:
Will do.  Thanks!

Commit:
[WebRTC] Extract BaseTarget-libaom.xcconfig from libaom.xcconfig
https://bugs.webkit.org/show_bug.cgi?id=267022
<rdar://120399649>

Reviewed by Elliott Williams.

* Source/ThirdParty/libwebrtc/Configurations/BaseTarget-libaom.xcconfig: Add.
- Extract settings from libaom.xcconfig.
(WARNING_CFLAGS):
- Add missing $(inherited) to define warning flags consistently.
- Add -Wno-unused-function to fix the build after adding $(inherited).
* Source/ThirdParty/libwebrtc/Configurations/libaom.xcconfig:
- Include BaseTarget-libaom.xcconfig after extracting settings.
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
- Add BaseTarget-libaom.xcconfig to the project.

Canonical link: https://commits.webkit.org/272641@main
